kore/src/Kore/Exec.hs

```diff
@@ -281,22 +292,32 @@ getExitCode
     :: (MonadIO simplifier, MonadSimplify simplifier)
     => VerifiedModule StepperAttributes
     -- ^ The main module
-    -> Pattern VariableName
+    -> OrPattern.OrPattern VariableName
     -- ^ The final configuration(s) of execution
     -> simplifier ExitCode
 getExitCode indexedModule finalConfig =
     takeExitCode $ \mkExitCodeSymbol -> do
         let mkGetExitCode t = mkApplySymbol (mkExitCodeSymbol []) [t]
-        exitCodePattern <-
-            Pattern.simplifyTopConfiguration (mkGetExitCode <$> finalConfig)
-            >>= Logic.scatter
-        case Pattern.term exitCodePattern of
-            Builtin_ (Domain.BuiltinInt (Domain.InternalInt _ exit))
-              | exit == 0 -> return ExitSuccess
-              | otherwise -> return $ ExitFailure $ fromInteger exit
-            _ -> return $ ExitFailure 111
+        exitCodePatterns <- do
+            pat <- Logic.scatter $ fmap mkGetExitCode `MultiOr.map` finalConfig
+            Pattern.simplifyTopConfiguration pat
```

Comment:
We need to gather the branches _here_, and remove `Logic.observeAllT` in `takeExitCode`, below. Otherwise, we would return `ExitFailure 111` when all the branches agree, if any branch would return `\bottom`.

```suggestion
        exitCodePatterns <-
            do
                pat <- Logic.scatter $ fmap mkGetExitCode `MultiOr.map` finalConfig
                Pattern.simplifyTopConfiguration pat
            & MultiOr.gather
```